src/chat/ChatScreen.js

```diff
@@ -23,6 +23,7 @@ import { getFetchingForNarrow } from './fetchingSelectors';
 import { getShownMessagesForNarrow, isNarrowValid as getIsNarrowValid } from './narrowsSelectors';
 import { addToOutbox } from '../actions';
 import { getAuth, getCaughtUpForNarrow } from '../selectors';
+import { getDraftForNarrow } from '../drafts/draftsSelectors';
```

Comment:
> It'd be good to keep the commit that removes `UNSAFE_componentWillReceiveProps` as focused as possible.

Well, here's one behavior change I've just found, and it's a nice one! Just before this commit, I get the following (and also on `master` I get something that looks the same) when I start editing a message, along with lots of lag. It goes away at this commit; I suspect that's because of removing `UNSAFE_componentWillReceiveProps`.

![image](https://user-images.githubusercontent.com/22248748/120737347-eb449180-c4bb-11eb-9548-ea225ef44f8b.png)

Reply:
> Hmm, could you say more about how adding `defaultTopic` and `defaultMessage` lets us get rid of `UNSAFE_componentWillReceiveProps`? What problems might we run into if we tried to get rid of it before adding those props? The changes look orthogonal to me, but I might be missing something. Or, if there would be problems, how about putting it after?

You're correct, these are orthogonal. I've refactored the commits to avoid mixing them up.